Update About (#4555)
woodpecker-ci/woodpecker: docs/docs/50-about.md

# About

Woodpecker is a community-driven open source CI/CD tool. It is designed to be lightweight, fast and simple to use. It can be used with many different Git providers and runners.

## History

> Woodpecker was originally forked by [@laszlocph](https://github.com/laszlocph) in 2019 from Drone 0.8 after licensing model was changed.

Reviewer: ```suggestion
Woodpecker was originally forked by [@laszlocph](https://github.com/laszlocph) in 2019 from Drone 0.8 after its licensing model was changed.
```

## Differences to Drone

> Woodpecker is a community-driven open source software published under Apache2 license that will always stay so. Drone CI is managed by [Harness](https://harness.io/) and available in two editions: [Open Source under Apache2 license and Enterprise under Polyform Small Business license](https://docs.drone.io/enterprise/#is-drone-open-source). In terms of functionality, Woodpecker is somewhere in between [Drone Enterprise and OSS](https://docs.drone.io/enterprise/#what-is-the-difference-between-open-source-and-enterprise).

Reviewer 2: ```suggestion
Woodpecker is a community-driven open source software published under Apache License 2.0 that will always stay so. Drone CI is managed by [Harness](https://harness.io/) and available in two editions: [Open Source under Apache License 2.0 and Enterprise under Polyform Small Business license](https://docs.drone.io/enterprise/#is-drone-open-source). In terms of functionality, Woodpecker is somewhere in between [Drone Enterprise and OSS](https://docs.drone.io/enterprise/#what-is-the-difference-between-open-source-and-enterprise).
```

> Woodpecker is a community-driven open source software published under Apache License 2.0 that will always stay so. Drone CI is managed by [Harness](https://harness.io/) and available in two editions: [Open Source under Apache License 2.0 and Enterprise under Polyform Small Business license](https://docs.drone.io/enterprise/#is-drone-open-source). In terms of functionality, Woodpecker is somewhere in between [Drone Enterprise and OSS](https://docs.drone.io/enterprise/#what-is-the-difference-between-open-source-and-enterprise).

Reviewer 3: I don't think including a subjective placement of where Woodpecker stands in contrast to Drone and Drone enterprise will age well. Leaving this out will make this part more static and less subjective.

Author: <details>
  <summary>Off topic</summary>

  I'm confused hearing about subjectivity from you. It is a little bit funny in the other hand.
</details>

> subjective placement of where Woodpecker stands in contrast to Drone

What exactly don't you agree with? Which (subjective) part, `1` or `2`?
1. Woodpecker is above Drone OSS.
   Hope, WP is not going to fall under this line.
3. Woodpecker is below Drone Enterprise.
   I'm afraid WP won't stand this fight now. But if you want... well... then start with objective features, not with complaining about my "subjectivity".

> I don't think including a subjective placement of where Woodpecker stands in contrast to Drone will age well

I would talk about it when Woodpecker becomes feature-competitive/comparable with Drone Enterprise.
_For now_ the sentence is right, if I understand you correctly. Besides, it won't be a problem to remove it _then_.

Secondly, there is a link to Drone OSS vs Enterprise comparison, which might give one a clue (until we have direct comparison table), what features Woodpecker might have.

Also, it is very rounded statement (like a circle 😄 ). I do not endorse or defame something, nor spreading the FUD.

All being said, there is no need to remove it, in my opinion.

Reviewer 4: > * Woodpecker is above Drone OSS.
>   Hope, WP is not going to fall under this line.
> 
> * Woodpecker is below Drone Enterprise.
>   I'm afraid WP won't stand this fight now. But if you want... well... then start with objective features, not with complaining about my "subjectivity".

Why do you tend to these controversial/provocative statements? If you prefer Drone, that's totally fine, just use it. If you prefer to use Woodpecker, cool, but in that case, help to improve it in a constructive way.

If you want to provide a full feature comparison matrix for Drone Woodpecker, feel free, but just adding "terms of functionality, Woodpecker is somewhere in between" is a what I call a subjective statement. Please remove this sentence for now, we can discuss this in a different place.

Author: I do not agree, that it is controversial. I think it is constructive. My preferences is subjective and is offtopic. I use Woodpecker, JFYI.

No, I do not want. I do not think it is subjective and gave some points why it should stay. You are only repeating "subjective", have not heard what is wrong with the sentence objectively.

`Allow edits by maintainers` is checked.

Reviewer 4: Alright then just let us know what feature Drone EE has that is missing in Woodpecker. And how do we count features that Woodpecker has which dont exist in Drone? One important example is the gatet pipeline in Woodpecker, such a feature does not exist till today even if we have requested it as former Drone CI user to Harness. So even if Woodpecker does not have all features that exist in Drone EE, Woodpecker has evolved as a dedicated CICD. So why do you even want to out it in a ranking with Drone CE and EE in a subjective way?

Author: > how do we count features that Woodpecker has which dont exist in Drone?

One by one. The same as with Drone.
Good example, thanks, +1 on the Woodpecker's side.
It would be great, if you/someone remembered more.

> Woodpecker has evolved as a dedicated CICD

That is amazing! I do not argue about it.

> Why do you even want to put it in a ranking with Drone CE and EE

I trimmed the end. Because, users are interested in differences, header is `Differences to Drone` and `there is a link to Drone OSS vs Enterprise comparison, which might give one a clue, what features Woodpecker have or might have`.

---

Today is a bit late and I'm not near the computer and won't be tomorrow. So, I will be able to stand for Drone on Monday.

Therefore, may I ask you for a favor? In order to not waste my time (and probably your/others), could you finish the PR (close or remove the sentence and merge) by then, _if you are not interested in such a "comparison" activity_?

Author: > gatet pipeline in Woodpecker
> +1 on the Woodpecker's side

I'm not sure, however. Drone has [Gated builds](https://drone.discourse.group/t/gated-builds-example-usage-help/10721/7) and [Signatures](https://docs.drone.io/signature/). Tie on this matter, I guess?

I've rounded the sentence even more. @pat-s, @xoxys does it suit you now?

Could someone provide two features, that do not exist in Drone in order to prove this addition? I would appreciate that.

Reviewer 4: > I'm not sure, however. Drone has [Gated builds](https://drone.discourse.group/t/gated-builds-example-usage-help/10721/7) and [Signatures](https://docs.drone.io/signature/). Tie on this matter, I guess?

This feature is only implemented half-way. Yes it can be used to protect against malicious changes to the `.drone.yml` but it only works for yaml not starlark and jsonnet. It also don't protect the CI from PR spam, which could result in some kind of DOS-ing.

> Could someone provide two features, that do not exist in Drone in order to prove this addition? I would appreciate that.

For me, the gated feature is still way more powerful in WP. The tight integration of the config linter especially in the Web UI is another unique feature.

Author: > implemented half-way

Depends on how you call the feature :)

> it can be used to protect against malicious changes to the .drone.yml

I thought about it, those security issues a couple of days before. If we call it as said in quote, it exist in Drone. Tie here.

It ismore convenient in Drone, I think.
Did you set `when.event:pull_request`? Good, your pipeline will run on PRs _automatically_.
Did someone change pipeline (to steal the secrets maybe)? No-no, admin (DevOps) reviews and approve first.

> PR spam,  DOS-ing

Yeah, though about it yesterday :) That is another feature. Indeed, there is no such thing in Drone. Seems, like +1 on WP side.

But how it is implemented in Woodpecker, makes me questioning it.
What is the purpose of setting `when.event:pull_request`? To run pipelines automatically. But you wait for approval on each one. So, I call it manually. Then don't set event and just run manually. This is also prevents from growing list of useless records in pipeline list. Example: this PR with applied suggestions.
I know, we cannot run pipeline manually from PR or Tag in Woodpecker. I wonder, if Drone can do this. If yes, then tie here too.

> starlark and jsonnet

2 points for Drone 👍 

> integration of the config linter

+1 for WP

> Signatures 

+1 for Drone

Reviewer 4: I'll give up on such discussions.

## Differences to Drone

> Woodpecker is a community-driven open source software published under Apache License 2.0 that will always stay so. Drone CI is managed by [Harness](https://harness.io/) and available in two editions: [Open Source under Apache License 2.0 and Enterprise under Polyform Small Business license](https://docs.drone.io/enterprise/#is-drone-open-source).
> In terms of Drone's feature-set coverage, Woodpecker is somewhere in between [Drone Enterprise and OSS](https://docs.drone.io/enterprise/#what-is-the-difference-between-open-source-and-enterprise), but also has a couple unique features.

Reviewer 4: ```suggestion
Woodpecker is a community-driven open source software published under the Apache License 2.0 and will always remain so. Drone CI is managed by [Harness](https://harness.io/) and is available in two editions: [Open Source under Apache License 2.0 and Enterprise under Polyform Small Business license](https://docs.drone.io/enterprise/#is-drone-open-source).
In terms of Drone's feature set, Woodpecker is somewhere between [Drone Enterprise and OSS](https://docs.drone.io/enterprise/#what-is-the-difference-between-open-source-and-enterprise), but also has some unique features.
```
